draft-cms-masque-ip-proxy-reqs.md

```diff
@@ -133,30 +143,33 @@ packets, in their unmodified entirety. The protocol will support both IPv6
 
 ## Maximum Transmission Unit
 
-The protocol will allow endpoints to negotiate the Maximum Transmission Unit
-(MTU) in use over a given Data Transport. This will allow avoiding IP
-fragmentation, especially as IPv6 does not allow IP fragmentation by nodes
+The protocol will allow endpoints to inform each other of the Maximum
+Transmission Unit (MTU) they are willing to forward. This will allow avoiding
+IP fragmentation, especially as IPv6 does not allow IP fragmentation by nodes
 along the path.
 
 ## IP Assignment
 
-Both the client or server may request to be assigned an IP address range. In
-response to the request, the peer will respond with an IP address range of its
-choosing.
+The server will be able to assign an IP address of its choosing to the client.
```

Comment:
I still believe that this interaction would work most naturally with the client requesting the IP address to assign. In lay terms, all I really want to say here is:

- If one of the endpoints wants an address, but doesn't have any opinion about what that address should be then it should be allowed to say "give me some address that you're able/willing to route to and I'll use it".
- If one of the endpoints has a set of addresses that it wants its peer to be able to route to, then it should be able to say "here's some addresses I'm able to route to" and the peer can agree or disagree to accept them.

Reply:
I agree that this is more natural, but having the ability to unilaterally offer an address could be useful?

Reply:
I think it could be too... but I worry that this introduces a lot of overlap with the route announcement approach, which I'm not sure how to reconcile.

Given that the unilateral offering could be rephrased into having the other peer request an address, it seems to afford us a feature that we can already achieve in other ways.

Reply:
Makes sense, I reverted this change and made address assignment request-response again.